containers/app/StorageSpaceStatus.js

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react';
+import { c } from 'ttag';
+import { CircularProgress, Dropdown, Icon, generateUID, useUser, usePopperAnchor } from 'react-components';
+import humanSize from 'proton-shared/lib/helpers/humanSize';
+
+const StorageSpaceStatus = () => {
+    const [{ MaxSpace, UsedSpace }] = useUser();
+    const [uid] = useState(generateUID('dropdown'));
+    const { anchorRef, isOpen, toggle, close } = usePopperAnchor();
+
+    const usedPercent = Math.round(UsedSpace / MaxSpace);
+    const maxSpaceFormatted = humanSize(MaxSpace);
+    const usedSpaceFormatted = humanSize(UsedSpace);
+
+    return (
+        <>
+            <CircularProgress
+                className="center"
+                progress={usedPercent}
+                aria-describedby={uid}
+                rootRef={anchorRef}
+                onClick={toggle}
+            >
+                <text
+                    x="50%"
+                    y="50%"
+                    fontFamily="Constantia"
+                    textAnchor="middle"
+                    alignmentBaseline="middle"
+                    stroke="#ffffff"
+                >
+                    i
+                </text>
+            </CircularProgress>
+            <span className="center opacity-40 smaller storage">{usedSpaceFormatted}</span>
+            <Dropdown
+                id={uid}
+                isOpen={isOpen}
+                anchorRef={anchorRef}
+                onClose={close}
+                originalPlacement="left-bottom"
+                auto
+            >
+                <div className="dropDown-content p1">
+                    <div className="absolute top-right mt0-5 mr0-5">
+                        <button className="flex flex-items-center">
+                            <Icon name="close" />
+                        </button>
+                    </div>
+                    <div className="flex">
+                        <div className="pr1 flex flex-items-center">
+                            <div className="relative">
+                                <CircularProgress progress={usedPercent} size={100} backgroundColor="#eeeeee" />
+                                <span className="centered-absolute">{usedPercent}%</span>
+                            </div>
+                        </div>
+                        <div className="w150p">
+                            <div className="big mt0 mb0">{c('Title').t`Storage`}</div>
+                            <div className="small color-black mt0 mb0">
+                                {c('Info').jt`${usedSpaceFormatted} of ${maxSpaceFormatted} used`}
+                            </div>
+                            <p className="smaller opacity-40 mt0 mb1">
+                                {c('Info').t`Your storage space is shared across all Proton products.`}
+                            </p>
+                            <a href="/settings/subscription" className="pm-button-blue">{c('Action').t`Upgrade`}</a>
```

Comment:
Should use `Link` if you are in the settings app, otherwise it'd refresh the page I think?

Reply:
Or perhaps each app should send a upgrade component to let each app define the behaviour themselves, since the URL will differ in the future

Reply:
Can we detect if the route goes inside the app or outside by checking `Routes` with `react-router`? Because, I'm thinking of using a component for it.

Reply:
I'm thinking about checking if the first part of the pathname matches the first part of the link.
```
    const [, app] = url.split('/');
    const [, currentApp] = location.pathname.split('/');

    if (app && currentApp === app) {
        return (
            <Link to={url} className={className} rel={rel}>
                {children}
            </Link>
        );
    }
```

Reply:
There will not be a 1-1 mapping between the pathname and the app so we need another solution

Reply:
I suggest passing a `upgradeUrl` or `<Upgrade>` component from the app